### PR TITLE
Migrate/teamcity

### PR DESCRIPTION
**Pull request by mvxt**
# Description
- Added migration page for migration from TeamCity.
- Fixed table width issue by adding custom css class and adding that class to the tables in migrate pages - does not affect other, currently existing tables
- Reordered competitor migration links in nav bar and intro page to be alphabetical order, easier to read.

# Reasons
Always good to have documents to help migrate from competitors. Had someone ask for it in the ideas forum as well: https://ideas.circleci.com/ideas/CCI-I-1276

### CODE REVIEW

**Comment by mvxt** [comment]
Fixed the table width issues with help from @originalhat . Thanks D!

Table fix

File: jekyll/assets/css/main.scss [hunk 1]
```diff
@@ -390,3 +390,10 @@ pre {
 .serveronly{
 	color: #4e9eb3; 
   }
+
+.table-migrate td, .table-migrate th {
+  width: 50%;
+  display: inline-block;
+  height: 100%;
+}
+
```
Comment (teesloane) [inline jekyll/assets/css/main.scss]: Nice work on this. Looks great on desktop. How do you feel about dropping the two into a column view on mobile?

image

**Comment by mvxt** [comment]
@teesloane Good catch on the mobile stuff. It looks like currently existing pages are side-scrolling in mobile view. Do we prefer single-column format on mobile? I'll post GIFs to compare the two views after I implement. Here's how horizontal scrolling looks:
H Scroll

**Comment by mvxt** [comment]
Nvm all, after testing the responsiveness it's all sorts of broken :( gotta work on fixing this.

**Comment by teesloane** [comment]
Hey @mvxt - Yeah; I deleted my original comment because I noticed that the other side-by-side tables weren't quite mobile ready, but it looks like it's already shipped. If you can get them running in a column on mobile ( I can't recall the scss vars off hand, but probably anything under 500~ px should be good...) 

Thanks for your hard work on this!

**Comment by mvxt** [comment]
@rosieyohannan Finished! The code side may not be the _**most**_ elegant. Maybe there's a way we can define a snippet and reuse in multiple places. But I saw that you defined snippets in a separate file and I thought that that was excessive for just a single page.

Responsiveness:
Responsive

**Comment by rosieyohannan** [comment]
This looks great @mvxt! I will do a full check over tomorrow 👍

File: jekyll/_cci2/migrating-from-teamcity.adoc [hunk 2]
```diff
@@ -0,0 +1,454 @@
...
+
+This document provides an overview of how to migrate from TeamCity to CircleCI.
+
+== Quick Start
+
+CircleCI has various differences in setup and configuration from TeamCity. For the best experience of trying CircleCI, we recommend getting started with the following:
+
+. **Read (or at least browse) this document** to solidify conceptual differences.
+. **Quickly build and run a simple https://circleci.com/docs/2.0/hello-world/#section=getting-started[Hello World job] in any repository**. This configuration demonstrates the simplicity of our YAML setup, and it showcases how to define both an execution environment and the steps to run within it. All jobs on CircleCI share this format.
+. **Once you understand the config, modify the executor and the individual steps to your needs.** For example, a https://circleci.com/docs/2.0/language-ruby/[Rails project] may use a Ruby container and run `bundler` commands. Consult our https://circleci.com/docs/2.0/tutorials/#section=configuration[Tutorials page] for various examples in different languages and frameworks.
```
Comment (rosieyohannan) [inline jekyll/_cci2/migrating-from-teamcity.adoc]:
```suggestion
. **Once you understand the config, modify the executor and the individual steps to meet your needs.** For example, a https://circleci.com/docs/2.0/language-ruby/[Rails project] may use a Ruby container and run `bundler` commands. Consult our https://circleci.com/docs/2.0/tutorials/#section=configuration[Tutorials page] for various examples in different languages and frameworks.
```

File: jekyll/_cci2/migrating-from-teamcity.adoc [hunk 3]
```diff
@@ -0,0 +1,454 @@
...
+. **After finishing one job, setup your other jobs on CircleCI and then https://circleci.com/docs/2.0/workflows/[configure a workflow]**. This would essentially migrate your Build Chain.
+. **Next, https://circleci.com/docs/2.0/optimizations/[optimize your jobs] with our advanced features** to see how much faster your builds could run on our platform.
```
Comment (rosieyohannan) [inline jekyll/_cci2/migrating-from-teamcity.adoc]:
```suggestion
. **Next, https://circleci.com/docs/2.0/optimizations/[optimize your jobs] with advanced features** to see how much faster your builds could run on the CircleCI platform.
```

File: jekyll/_cci2/migrating-from-teamcity.adoc [hunk 4]
```diff
@@ -0,0 +1,454 @@
...
+. **Throughout this process, consult our https://circleci.com/docs/2.0/configuration-reference/#section=configuration[Config Reference]** for a list of all available configuration on our platform.
```
Comment (rosieyohannan) [inline jekyll/_cci2/migrating-from-teamcity.adoc]:
```suggestion
. **Throughout this process, consult the https://circleci.com/docs/2.0/configuration-reference/#section=configuration[Configuration Reference]** for a list of all available configuration on the CircleCI platform.
```

File: jekyll/_cci2/migrating-from-teamcity.adoc [hunk 5]
```diff
@@ -0,0 +1,454 @@
...
+
+== Why Migrate to CircleCI?
+
+CircleCI is a modern CI/CD tool aimed at performance and developer productivity. With a flexible system for configuration, you can define complex workflows and jobs for your application's specific needs.
+
+**Configuration As Code**:: The majority of CircleCI's configuration is done via YAML. It is easy to maintain and trivial to learn, and your CI/CD pipeline can be tracked and versioned like the rest of your source code.
```
Comment (rosieyohannan) [inline jekyll/_cci2/migrating-from-teamcity.adoc]: Just wondering what the significance of double "::" is?

File: jekyll/_cci2/migrating-from-teamcity.adoc [hunk 6]
```diff
@@ -0,0 +1,454 @@
...
+**Shareable, Reusable Configuration**:: Steps, execution environments, and jobs can be templated, parameterized, shared, and reused across different projects and teams using https://circleci.com/docs/2.0/orb-intro/#section=configuration[CircleCI orbs]. Take advantage of the https://circleci.com/orbs/registry/[numerous integrations and orbs] already written for you.
+
+We have various other features that set our solution apart. https://circleci.com/signup/[Sign up for a free account today] and try us out, or if you are interested in CircleCI for your team, https://circleci.com/talk-to-us/?source-button=MigratingFromGithubActionsDoc[contact our sales team] to set up a trial.
+
+== Concepts
+
+=== High-Level Differences
+
+* CircleCI and TeamCity share the concept of projects.
+* TeamCity's Build Configurations are equivalent to https://circleci.com/docs/2.0/concepts/#jobs[CircleCI's jobs], which are atomic elements of work consisting of many steps or commands to achieve some sort of goal (e.g. run unit tests).
```
Comment (rosieyohannan) [inline jekyll/_cci2/migrating-from-teamcity.adoc]:
```suggestion
* TeamCity's Build Configurations are equivalent to https://circleci.com/docs/2.0/concepts/#jobs[CircleCI's jobs], which are atomic elements of work consisting of many steps or commands to achieve a goal (e.g. run unit tests).
```

File: jekyll/_cci2/migrating-from-teamcity.adoc [hunk 7]
```diff
@@ -0,0 +1,454 @@
...
+----
+|===
+
+For more information on CircleCI Concepts, visit our https://circleci.com/docs/2.0/concepts/[Concepts] and https://circleci.com/docs/2.0/pipelines/#section=pipelines[Pipelines] documentation pages.
+
+== Configuration
+
+=== Environment
+
+TeamCity requires setting up a build agent with the required O.S. & tools installed and a corresponding Build Configuration. In CircleCI, all job configurations have an Executor definition, and CircleCI handles spinning up said agents for you. See our list of https://circleci.com/docs/2.0/executor-types/[available executors].
```
Comment (rosieyohannan) [inline jekyll/_cci2/migrating-from-teamcity.adoc]:
```suggestion
TeamCity requires setting up a build agent with the required OS and tools installed and a corresponding Build Configuration. In CircleCI, all job configurations have an Executor definition, and CircleCI handles spinning up said agents for you. See our list of https://circleci.com/docs/2.0/executor-types/[available executors].
```

**Comment by teesloane** [comment]
Nice work on the responsive tables @mvxt  - looks good to me.

File: jekyll/_cci2/migrating-from-teamcity.adoc (hunk 5 above)
Comment (mvxt) [reply jekyll/_cci2/migrating-from-teamcity.adoc]: Ah oops that's a typo.

Comment (mvxt) [reply jekyll/_cci2/migrating-from-teamcity.adoc]: @rosieyohannan Oh actually, no it's not a typo. The double colons are description lists in ASCII Doc: https://asciidoctor.org/docs/asciidoc-writers-guide/#description-lists

Comment (rosieyohannan) [reply jekyll/_cci2/migrating-from-teamcity.adoc]: Oh great! :-D

**Comment by mvxt** [comment]
Hey @rosieyohannan , did you ever get the chance to look at this? I'll merge latest master into this branch tomorrow when I get into the office, but I was wondering how everything else looked content-wise. Let me know if there's anything else you'd like me to adjust.

**Comment by rosieyohannan** [comment]
@mvxt yes sorry for the delay, it all looks great! thanks so much! Also thanks for alphabetising the guides too 👏 🎉 :shipit:

**Pull request merged**